### PR TITLE
TELEMETRY_MOD=14051_SWAPPED

### PR DESCRIPTION
**Pull request by xmxwx**
TELEMETRY_MOD=14051 (by gerardv @ forum.turnigy9xr.com) makes USART0 available
by reallocating AIL and THR onto pins 14 and 15 of the 14051 multiplexer.

TELEMETRY_MOD=14051_SWAPPED is based on TELEMETRY_MOD=14051, but uses USART1
instead of USART0, freeing USART0 for other purposes (debugging, interfacing
via AVRISP port). USART1 is freed up by reallocating LVert Trim Dn/Up onto
14051's 12 and 1 pins respectively.

Additionally, this commit cleans a few things up: moves code related to this
telemetry mod to board_stock and replaces a few magic numbers with enums.

### CODE REVIEW

File: radio/src/opentx.cpp [hunk 1]
```diff
@@ -1514,13 +1514,10 @@ void getADC()
 /**
  * Read ADC using 10 bits
  */
-inline uint16_t read_adc10(uint8_t adc_input)
```
Comment (bsongis) [inline radio/src/opentx.cpp]: Doesn't it use more flash?

File: radio/src/opentx.cpp [hunk 2]
```diff
@@ -1514,13 +1514,10 @@ void getADC()
 /**
  * Read ADC using 10 bits
  */
-inline uint16_t read_adc10(uint8_t adc_input) 
+uint16_t read_adc10(uint8_t adc_input)
```
Comment (bsongis) [inline radio/src/opentx.cpp]: Not related to your changes, but now you are on it ... this part of code should be in the driver! Perhaps an adc_driver.cpp with consistent function name adcRead(...)

File: radio/src/opentx.cpp [hunk 3]
```diff
@@ -2994,14 +2939,14 @@ ISR(TIMER3_CAPT_vect) // G: High frequency noise can cause stack overflo with IS
...
-FORCEINLINE void DSM2_USART0_vect()
+FORCEINLINE void DSM2_USART_vect()
 {
   UDR0 = *((uint16_t*)pulses2MHzRPtr); // transmit next byte
 
   pulses2MHzRPtr += sizeof(uint16_t);
 
   if (pulses2MHzRPtr == pulses2MHzWPtr) { // if reached end of DSM2 data buffer ...
-    UCSR0B &= ~(1 << UDRIE0); // disable UDRE0 interrupt
+    UCSRB_N(TLM_USART) &= ~(1 << UDRIE_N(TLM_USART)); // disable UDRE interrupt
```
Comment (bsongis) [inline radio/src/opentx.cpp]: Seems also driver code!

File: radio/src/opentx.h [hunk 4]
```diff
@@ -1751,4 +1751,75 @@ void varioWakeup();
...
+#define RXD_PORT0 PORTE
+#define RXD_PORT_PIN0 PORTE0
+
+#if defined(TELEMETRY_MOD_14051_SWAPPED)
+#define TLM_USART 1
+#else
+#define TLM_USART 0
+#endif
+
+#define _DOR_N(usart_no) DOR ## usart_no
```
Comment (bsongis) [inline radio/src/opentx.h]: All these definitions in the driver code!

File: radio/src/targets/stock/board_stock.cpp [hunk 5]
```diff
@@ -242,21 +289,29 @@ bool switchState(EnumKeys enuk)
...
-  1<<INP_D_TRM_LH_DWN,  // bit 7
-  1<<INP_D_TRM_LH_UP,
-  1<<INP_D_TRM_LV_DWN,
-  1<<INP_D_TRM_LV_UP,
-  1<<INP_D_TRM_RV_DWN,
-  1<<INP_D_TRM_RV_UP,
-  1<<INP_D_TRM_RH_DWN,
-  1<<INP_D_TRM_RH_UP    // bit 0
-};
+uint8_t trim_helper(uint8_t neg_pind, uint8_t idx)
```
Comment (bsongis) [inline radio/src/targets/stock/board_stock.cpp]: We try to use lowerCamelCase in new code!

**Comment by bsongis** [comment]
Some cosmetics here and there, but it's a nice one, thanks!

**Comment by xmxwx** [comment]
@bsongis thanks for the feedback, I really appreciate it!

As suggested, I moved most of the ADC-related code to adc_driver.cpp. This nicely matches adc_driver.cpp done in ARM-based boards.

Cleaning up the USART code (which grew a bit due to my attempts to make it more agile) is still TODO. I'll try to do it within this pull request.

**Comment by xmxwx** [comment]
OK, I've integrated a few the remaining remarks and manually rebased it to the next branch (I should've develop on next from the start!). Force-pushing it here, however, is not a good idea since GitHub doesn't interpret it correctly by changing the base branch. Therefore, let's close it, and open another pull request.

**Pull request closed without merging**